Lib/pydoc.py

```diff
@@ -488,28 +488,33 @@ def fail(self, object, name=None, *args):
     def getdocloc(self, object, basedir=sysconfig.get_path('stdlib')):
         """Return the location of module docs or None"""
 
+        if not inspect.ismodule(object):
+            return None
+
+        try:
+            from pydoc_data.topics import modules
+        except ImportError:
+            return None
+
+        if object.__name__ not in modules:
+            return None
+
         try:
             file = inspect.getabsfile(object)
         except TypeError:
             file = '(built-in)'
 
+        if 'site-packages' in file:
+            # XXX: Maybe we could add a hook for including a documentation URL?
+            return None
+
         docloc = os.environ.get("PYTHONDOCS", self.PYTHONDOCS)
+        html_file = modules[object.__name__] + '.html'
+
+        if docloc.startswith('http'):
```

Comment:
If this line intends to catch HTTP links, I think the previous code had a better test:

```suggestion
        if docloc.startswith(('http://', 'https://')):
```